internal/internal_worker_base.go

```diff
@@ -421,3 +424,20 @@ func (bw *baseWorker) Stop() {
 	}
 	return
 }
+
+func getAutoConfigHint(task interface{}) *shared.AutoConfigHint {
+	switch t := task.(type) {
+	case *workflowTask:
+		if t.task != nil {
+			return t.task.AutoConfigHint
+		}
+		return t.autoConfigHint
+	case *activityTask:
+		if t.task != nil {
+			return t.task.AutoConfigHint
+		}
+		return t.autoConfigHint
+	default:
+		return nil
+	}
```

Comment:
instead of this switch case (which is not future proof), we can cast the task to `autoConfigHintAwareTask` interface and get the auto config hint